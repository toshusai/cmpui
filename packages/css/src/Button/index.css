.cmpui_button {
  background-color: var(--cmpui-primary-color);

  align-items: center;
  border-radius: 9999px;
  border: none;
  box-sizing: border-box;
  color: white;
  cursor: pointer;
  display: flex;
  font-family: unset;
  font-size: 16px;
  height: 32px;
  justify-content: center;
  line-height: 16px;
  outline: none;
  padding: 0;
  padding: 8px 16px;
  text-align: center;
  text-decoration: none;
}

.cmpui_button:hover {
  background-color: var(--cmpui-primary-hover-color);
}

.cmpui_button:active {
  background-color: var(--cmpui-primary-active-color);
}

.cmpui_button:focus-visible {
  outline: var(--cmpui-primary-color) solid 1px;
  outline-offset: 1px;
}

/* disabled */

.cmpui_button:disabled {
  cursor: default;
  opacity: 0.5;
}

.cmpui_button:disabled:hover {
  background-color: var(--cmpui-primary-color);
}

.cmpui_button:disabled:active {
  background-color: var(--cmpui-primary-color);
}

/* Variant: secondary */

.cmpui_button[data-variant="secondary"] {
  background-color: var(--cmpui-secondary-color);
}

.cmpui_button[data-variant="secondary"]:hover {
  background-color: var(--cmpui-secondary-hover-color);
}

.cmpui_button[data-variant="secondary"]:active {
  background-color: var(--cmpui-secondary-active-color);
}

.cmpui_button[data-variant="secondary"]:disabled:hover {
  background-color: var(--cmpui-secondary-color);
}

.cmpui_button[data-variant="secondary"]:disabled:active {
  background-color: var(--cmpui-secondary-color);
}

/* Size: S */

.cmpui_button[data-size="S"] {
  font-size: 12px;
  height: 24px;
  line-height: 12px;
  padding: 4px 12px;
}
