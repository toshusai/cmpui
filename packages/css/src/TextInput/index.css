.cmpui_text-input__root {
  background-color: var(--cmpui-bg-color);
  border: 1px solid var(--cmpui-border-color);

  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  display: flex;
  height: 32px;
  justify-content: flex-start;
  position: relative;
  text-decoration: none;
}

.cmpui_text-input__root:focus-within {
  outline: var(--cmpui-primary-color) solid 1px;
  outline-offset: 1px;
}

.cmpui_text-input__root[aria-disabled="true"] {
  cursor: default;
  opacity: 0.5;
}

.cmpui_text-input__root[aria-invalid="true"] {
  border: 1px solid var(--cmpui-error-color);
}
.cmpui_text-input__root[aria-invalid="true"]:hover {
  border: 1px solid var(--cmpui-error-color);
}

.cmpui_text-input__root:hover {
  border: 1px solid var(--cmpui-border-hover-color);
}

.cmpui_text-input__root[aria-disabled="true"]:hover {
  border: 1px solid var(--cmpui-border-color);
}

.cmpui_text-input__label {
  background-image: linear-gradient(
    to bottom,
    transparent 10%,
    var(--cmpui-bg-color) 40%,
    var(--cmpui-bg-color) 60%,
    transparent 60%
  );
  color: var(--cmpui-text2-color);

  font-size: 12px;
  line-height: 14px;
  padding: 0 4px;
  position: absolute;
  transform: translate(4px, -16px);
}
.cmpui_text-input__root[aria-invalid="true"] > .cmpui_text-input__label {
  color: var(--cmpui-error-color);
}

.cmpui_text-input__input:nth-child(3) {
  padding-left: 0px;
}

.cmpui_text-input__input {
  color: var(--cmpui-text-color);
  font-family: var(--cmpui-font-family);

  background: transparent;
  border: none;
  box-sizing: border-box;
  display: flex;
  font-size: 16px;
  height: 100%;
  line-height: 16px;
  outline: none;
  padding: 0 8px;
  width: 100%;
}

.cmpui_text-input__input::placeholder {
  color: var(--cmpui-text3-color);
  font-style: italic;
}
