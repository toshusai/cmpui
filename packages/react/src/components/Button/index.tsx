import { forwardRef } from "react";
import { CustomJSXElement } from "../../types/CustomJSXElement";
import { classNames } from "../../utils/classNames";

export type ButtonProps<T extends CustomJSXElement = "button"> = {
  as?: T;
  size?: "S" | "M";
  variant?: "primary" | "secondary";
} & Omit<React.ComponentProps<T>, "as">;

export const Button = forwardRef(function Button<T extends CustomJSXElement>(
  {
    variant = "primary",
    size = "M",
    children,
    className,
    as,
    ...props
  }: ButtonProps<T>,
  ref: React.ForwardedRef<HTMLButtonElement>,
) {
  const Component = as || "button";
  return (
    <Component
      ref={ref}
      type="button"
      className={classNames("cmpui_button", className)}
      {...props}
      data-variant={variant}
      data-size={size}
    >
      {children}
    </Component>
  );
}) as <T extends CustomJSXElement = "button">(p: ButtonProps<T>) => JSX.Element;
