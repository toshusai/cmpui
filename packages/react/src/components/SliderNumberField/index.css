.cmpui_slider-number-input__root:focus {
  cursor: text;
}

.cmpui_slider-number-input__label {
  cursor:
    url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgLTk2MCA5NjAgOTYwIiB3aWR0aD0iMjQiPgogICAgPGRlZnM+CiAgICAgICAgPGZpbHRlciBpZD0ic2hhZG93Ij4KICAgICAgICAgICAgPGZlRHJvcFNoYWRvdyBkeD0iMCIgZHk9IjAiIHN0ZERldmlhdGlvbj0iMzIiCiAgICAgICAgICAgICAgICBmbG9vZC1jb2xvcj0iYmxhY2siCiAgICAgICAgICAgICAgICBmbG9vZC1vcGFjaXR5PSIxIgogICAgICAgICAgICAvPgogICAgICAgIDwvZmlsdGVyPgogICAgPC9kZWZzPgogICAgPHBhdGgKICAgICAgICBzdHlsZT0iZmlsdGVyOnVybCgjc2hhZG93KTsiCiAgICAgICAgc3Ryb2tlPSJ3aGl0ZSIgc3Ryb2tlLXdpZHRoPSI0OCIKICAgICAgICBkPSJNMjkwLjc0LTI5MC4wNDIgMTAwLjc4Mi00ODAgMjkwLjc0LTY2OS45NThsNzQuNTIyIDc0LjY1My02MS43MzkgNjIuMzA0aDM1Mi45NTRsLTYxLjMwNS02Mi44NjkgNzQuMDg4LTc0LjA4OEw4NTkuMjE4LTQ4MCA2NjkuODI2LTI5MC42MDdsLTc0LjY1NC03NC4wODggNjEuMzA1LTYyLjMwNEgzMDIuOTU4bDYyLjQzNSA2Mi4zMDQtNzQuNjUzIDc0LjY1M1oiIC8+Cjwvc3ZnPg==)
      8 8,
    ew-resize;
}
