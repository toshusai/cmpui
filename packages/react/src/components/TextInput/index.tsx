import { forwardRef, memo, useRef } from "react";

import { mergeRefs } from "../..";
import { classNames } from "../../utils/classNames";
import { useNativeOnChange } from "../../utils/interactions/useNativeOnChange";

import "./index.css";

export type TextInputProps = {
  htmlPrefix?: string;
  invalid?: boolean;
  label?: React.ReactNode;
  placeholder?: string;
  prefix?: React.ReactNode;
  suffix?: React.ReactNode;
  value?: string;
  onNativeChange?: (e: Event) => void;
} & Omit<React.InputHTMLAttributes<HTMLInputElement>, "prefix">;

export const TextInput = memo(
  forwardRef<HTMLInputElement, TextInputProps>(function TextInput(
    {
      className,
      htmlPrefix,
      invalid,
      label,
      placeholder,
      prefix,
      suffix,
      value,
      onNativeChange,
      ...props
    },
    fRef,
  ) {
    const ref = useRef<HTMLInputElement>(null);
    useNativeOnChange(ref, onNativeChange);
    return (
      <div
        className={classNames("cmpui_text-input__root", className)}
        aria-disabled={props.disabled}
        aria-invalid={invalid}
      >
        {prefix}
        {typeof label === "string" ? (
          <label className="cmpui_text-input__label" htmlFor={props.id}>
            {label}
          </label>
        ) : (
          label
        )}
        <input
          ref={mergeRefs(ref, fRef)}
          type="text"
          className="cmpui_text-input__input"
          value={value ?? ""}
          placeholder={placeholder ?? ""}
          prefix={htmlPrefix}
          aria-invalid={invalid}
          {...props}
        />
        {suffix}
      </div>
    );
  }),
);
