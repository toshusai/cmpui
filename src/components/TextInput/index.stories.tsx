import type { Meta, StoryObj } from "@storybook/react";
import { useState } from "react";
import { TextInput } from ".";

const meta: Meta<typeof TextInput> = {
  title: "Components/Form/TextInput",
  component: TextInput,
  args: {
    label: "Label",
  },
};

export default meta;
type Story = StoryObj<typeof TextInput>;

export const Basic: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        placeholder="Placeholder"
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const Label: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        id="text-input-label"
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const Placeholder: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label=""
        placeholder="Placeholder"
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const Suffix: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        prefix={
          <svg
            xmlns="http://www.w3.org/2000/svg"
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            style={{
              maxWidth: 16,
              height: 16,
            }}
          >
            <title>search</title>
            <path d="M784-120 532-372q-30 24-69 38t-83 14q-109 0-184.5-75.5T120-580q0-109 75.5-184.5T380-840q109 0 184.5 75.5T640-580q0 44-14 83t-38 69l252 252-56 56ZM380-400q75 0 127.5-52.5T560-580q0-75-52.5-127.5T380-760q-75 0-127.5 52.5T200-580q0 75 52.5 127.5T380-400Z" />
          </svg>
        }
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const Disabled: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        value={value}
        onChange={(event) => setValue(event.target.value)}
        disabled
      />
    );
  },
};

export const Invalid: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        value={value}
        onChange={(event) => setValue(event.target.value)}
        invalid
      />
    );
  },
};
