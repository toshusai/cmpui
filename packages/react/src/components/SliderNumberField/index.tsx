import { forwardRef, memo, useCallback, useMemo, useState } from "react";

import { classNames } from "../..";
import { TextInput } from "../TextInput";
import { createWindowDragHandler } from "../TreeView/lib/createWindowDragHandler";

import { parseCursorStyle } from "./lib/parseCursorStyle";
import { inputEval } from "./lib/inputEval";

import "./index.css";

export type SliderNumberField = {
  label: string;
  step?: number;
  value?: number[];
  onChangeValue?: (value: number[]) => void;
  numToString?: (value: number) => string;
  evalFunc?: (str: string, mixed?: number) => number;
  suffix?: React.ReactNode;
  onStart?: () => void;
  onEnd?: () => void;
} & Omit<React.InputHTMLAttributes<HTMLInputElement>, "value">;

export const SliderNumberField = memo(
  forwardRef<HTMLInputElement, SliderNumberField>(function SliderNumberField(
    {
      label,
      value,
      step = 1,
      suffix,
      className,
      numToString = (value) => value.toString(),
      onChangeValue,
      evalFunc = inputEval,
      onStart,
      onEnd,
      ...props
    },
    ref,
  ) {
    const isMixed =
      Array.isArray(value) &&
      value.reduce((acc, v) => {
        return acc || v !== value[0];
      }, false);

    const [_value, set_value] = useState<string>(
      isMixed ? "mixed" : typeof value === "number" ? numToString(value) : "",
    );

    const [isFocused, setIsFocused] = useState(false);

    const down = useMemo(
      () =>
        createWindowDragHandler<
          | {
              startValue: number | number[];
              startX?: number;
              el?: HTMLImageElement;
              input?: HTMLInputElement;
            }
          | undefined,
          | {
              value?: number | number[];
            }
          | undefined
        >({
          onDown: (e) => {
            if (isFocused && e.currentTarget instanceof HTMLInputElement)
              return;
            document.body.requestPointerLock();

            const el = document.createElement("img");
            el.style.position = "fixed";
            document.body.appendChild(el);

            if (e.currentTarget instanceof HTMLElement) {
              const cursor = getComputedStyle(e.currentTarget).cursor;
              const result = parseCursorStyle(cursor);
              if (result) {
                el.src = result.src;
                // 0.5 for visual adjustment
                el.style.left = `${e.clientX - Number(result.x) - 0.5}px`;
                el.style.top = `${e.clientY - Number(result.y) - 0.5}px`;
                document.body.appendChild(el);
              }
            }

            onStart?.();
            return {
              startValue: value ?? 0,
              startX: e.clientX,
              el,
              input: e.currentTarget as HTMLInputElement,
            };
          },
          onMove: (e, ctx, prev) => {
            if (!ctx) return;
            const diff = e.movementX;
            const diffY = e.movementY;
            const el = ctx.el;
            if (!el) return;
            const left = Number(el.style.left.replace("px", ""));
            const offset = 12;
            if (left + diff < -offset) {
              el.style.left = `${window.innerWidth + diff - offset}px`;
            } else if (left + diff > window.innerWidth - offset) {
              el.style.left = `${diff - offset}px`;
            } else {
              el.style.left = `${left + diff}px`;
            }

            const top = Number(el.style.top.replace("px", ""));
            if (top + diffY < -offset) {
              el.style.top = `${window.innerHeight + diffY - offset}px`;
            } else if (top + diffY > window.innerHeight - offset) {
              el.style.top = `${diffY - offset}px`;
            } else {
              el.style.top = `${top + diffY}px`;
            }

            if (value !== undefined) {
              let v: number[] = value;
              if (prev?.value !== undefined && Array.isArray(prev.value)) {
                v = prev.value;
              }
              const newValues = v.map((value) => {
                return value + Math.round(diff) * step;
              });
              onChangeValue?.(newValues);
              return {
                value: newValues,
              };
            }
          },
          onUp: (_, ctx) => {
            if (!ctx) return;
            ctx.el?.remove();
            document.exitPointerLock();
            onEnd?.();
          },
        }),
      [isFocused, onChangeValue, onStart, onEnd, step, value],
    );

    const handleCommit = useCallback(
      (rawValue: string) => {
        if (value !== undefined) {
          const newValues = value.map((value) => {
            return evalFunc(rawValue, value);
          });
          if (newValues.some((value) => Number.isNaN(value))) return value;
          onChangeValue?.(newValues);
          return newValues;
        }
      },
      [evalFunc, onChangeValue, value],
    );

    const handleBlur = useCallback(
      (e: React.FocusEvent<HTMLInputElement>) => {
        handleCommit(e.currentTarget.value);
        setIsFocused(false);
      },
      [handleCommit],
    );

    const handleChange = useCallback(
      (e: React.ChangeEvent<HTMLInputElement>) => {
        const rawValue = e.currentTarget.value;
        set_value(rawValue);
      },
      [],
    );

    const handleClick = useCallback(
      (e: React.MouseEvent<HTMLInputElement, MouseEvent>) => {
        e.currentTarget.select();
      },
      [],
    );

    const getValue = useMemo(() => {
      if (isFocused) {
        return _value;
      }
      if (isMixed) {
        return "mixed";
      }
      if (value === undefined) {
        return "";
      }
      if (typeof value === "number") {
        return numToString(value);
      }
      return numToString(value[0]);
    }, [_value, isFocused, isMixed, numToString, value]);

    const handleFocus = useCallback(
      (e: React.FocusEvent<HTMLInputElement>) => {
        e.currentTarget.select();
        set_value(getValue);
        setIsFocused(true);
      },
      [getValue],
    );

    const handleKeyDown = useCallback(
      (e: React.KeyboardEvent<HTMLInputElement>) => {
        if (e.key === "Enter") {
          const newValues = handleCommit(e.currentTarget.value);
          if (newValues === undefined) {
            return;
          }
          if (typeof newValues === "number") {
            set_value(newValues ? numToString(newValues) : "");
          } else {
            const isMixed = newValues.reduce((acc, value) => {
              return acc || value !== newValues[0];
            }, false);
            if (isMixed) {
              set_value("mixed");
            } else {
              set_value(numToString(newValues[0]));
            }
          }
        } else if (e.key === "Escape") {
          set_value(getValue);
        } else if (e.key === "ArrowUp") {
          if (value === undefined) {
            return;
          }

          const isMixed = value.reduce((acc, v) => {
            return acc || v !== value[0];
          }, false);
          if (isMixed) {
            onChangeValue?.(value.map((value) => value + step));
          } else {
            set_value(numToString(value[0] + step));
          }
          e.preventDefault();
        } else if (e.key === "ArrowDown") {
          if (value === undefined) {
            return;
          }

          const isMixed = value.reduce((acc, v) => {
            return acc || v !== value[0];
          }, false);
          if (isMixed) {
            onChangeValue?.(value.map((value) => value - step));
          } else {
            set_value(numToString(value[0] - step));
          }
          e.preventDefault();
        }
      },
      [getValue, handleCommit, numToString, onChangeValue, step, value],
    );

    return (
      <TextInput
        {...props}
        ref={ref}
        className={classNames("cmpui_slider-number-input__root", className)}
        onFocus={handleFocus}
        onKeyDown={handleKeyDown}
        onBlur={handleBlur}
        onChange={handleChange}
        onClick={handleClick}
        value={getValue}
        prefix={suffix}
        label={
          <label
            onPointerDown={down}
            className="cmpui_text-input__label cmpui_slider-number-input__label"
          >
            {label}
          </label>
        }
      />
    );
  }),
);
