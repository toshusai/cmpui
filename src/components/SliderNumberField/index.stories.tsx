import type { Meta, StoryObj } from "@storybook/react";
import { useState } from "react";
import { SliderNumberField } from ".";
import { TextInput } from "../TextInput";

const meta: Meta<typeof SliderNumberField> = {
  title: "Form/SliderNumberField",
  component: SliderNumberField,
};

export default meta;

type Story = StoryObj<typeof SliderNumberField>;

export const Basic: Story = {
  render: function Render(props) {
    const [value, setValue] = useState([0]);
    return (
      <SliderNumberField
        {...props}
        label="value"
        inputMode="decimal"
        value={value}
        onChangeValue={setValue}
      />
    );
  },
};

export const Mixed: Story = {
  render: function Render(props) {
    const [values, setValues] = useState([1, 2, 3]);
    return (
      <div>
        <SliderNumberField
          {...props}
          label="value"
          value={values}
          onChangeValue={(values) => {
            if (!Array.isArray(values)) throw new Error("values is not array");
            setValues(values);
          }}
        />
        <div
          style={{
            display: "flex",
            flexDirection: "column",
            gap: 4,
            marginTop: 8,
          }}
        >
          {values.map((value, index) => (
            <TextInput
              type="number"
              key={index}
              value={value.toString()}
              onChange={(e) => {
                const newValues = [...values];
                newValues[index] = Number(e.target.value);
                setValues(newValues);
              }}
            />
          ))}
        </div>
      </div>
    );
  },
};
