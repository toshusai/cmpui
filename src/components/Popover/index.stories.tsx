import { Meta, StoryObj } from "@storybook/react";
import { useCallback, useState } from "react";
import { Popover } from ".";
import { Button } from "../Button";
import { SliderNumberField } from "../SliderNumberField";
import { TextInput } from "../TextInput";

const meta: Meta<typeof Popover> = {
  title: "Components/Float/Popover",
  component: Popover,
};

export default meta;
type Story = StoryObj<typeof Popover>;

export const Primary: Story = {
  render: function RenderBasic() {
    const [open, setOpen] = useState(false);
    return (
      <>
        <Popover
          isOpen={open}
          onOpenChange={() => setOpen(false)}
          content={
            <Body
              onClose={() => {
                setOpen(false);
              }}
            />
          }
        >
          <Button onClick={() => setOpen(true)}>Open Popover</Button>
        </Popover>
      </>
    );
  },
};

function Body({
  onClose,
}: {
  onClose?: () => void;
} = {}) {
  const [name, setName] = useState("Awesome project");
  const [width, setWidth] = useState(256);
  const [height, setHeight] = useState(512);

  const handleChangeName = useCallback(
    (e: React.ChangeEvent<HTMLInputElement>) => {
      setName(e.target.value);
    },
    [],
  );

  const handleChangeWidth = useCallback((value: number[]) => {
    setWidth(value[0]);
  }, []);

  const handleChangeHeight = useCallback((value: number[]) => {
    setHeight(value[0]);
  }, []);

  return (
    <div
      style={{
        textAlign: "center",
        padding: 16,
        display: "flex",
        flexDirection: "column",
        gap: 12,
        width: 256,
      }}
    >
      <TextInput
        placeholder="input"
        label="name"
        value={name}
        onChange={handleChangeName}
        autoFocus
      />
      <div style={{ display: "flex", gap: 8 }}>
        <SliderNumberField
          placeholder="input"
          label="width"
          value={[width]}
          onChangeValue={handleChangeWidth}
        />
        <div
          style={{
            display: "flex",
            alignItems: "center",
          }}
        >
          Ã—
        </div>
        <SliderNumberField
          placeholder="input"
          label="heigt"
          value={[height]}
          onChangeValue={handleChangeHeight}
        />
      </div>

      <Button onClick={onClose}>Save</Button>
    </div>
  );
}
