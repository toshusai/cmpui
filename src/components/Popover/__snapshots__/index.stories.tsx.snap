// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Float/Popover Primary smoke-test 1`] = `
<button type="button"
        class="cmpui_button"
        data-variant="primary"
        data-size="M"
>
  Open Popover
</button>
`;
